c10/cuda/CUDACachingAllocator.cpp

```diff
@@ -2151,6 +2149,46 @@ class DeviceCachingAllocator {
     return blocks;
   }
 
+  void maybe_make_new_pool_and_inc_use_count(MempoolId_t mempool_id) {
```

Comment:
This name is confusing, because while you might make a new pool, you ALWAYS inc the use count. What about... `ensure_exists_and_incref_pool`